src/unix/linux_like/android/b32/arm.rs

```diff
@@ -1,5 +1,128 @@
 pub type c_char = u8;
 pub type wchar_t = u32;
+pub type greg_t = i32;
+
+s! {
+    pub struct sigcontext {
+        pub trap_no: ::c_ulong,
+        pub error_code: ::c_ulong,
+        pub oldmask: ::c_ulong,
+        pub arm_r0: ::c_ulong,
+        pub arm_r1: ::c_ulong,
+        pub arm_r2: ::c_ulong,
+        pub arm_r3: ::c_ulong,
+        pub arm_r4: ::c_ulong,
+        pub arm_r5: ::c_ulong,
+        pub arm_r6: ::c_ulong,
+        pub arm_r7: ::c_ulong,
+        pub arm_r8: ::c_ulong,
+        pub arm_r9: ::c_ulong,
+        pub arm_r10: ::c_ulong,
+        pub arm_fp: ::c_ulong,
+        pub arm_ip: ::c_ulong,
+        pub arm_sp: ::c_ulong,
+        pub arm_lr: ::c_ulong,
+        pub arm_pc: ::c_ulong,
+        pub arm_cpsr: ::c_ulong,
+        pub fault_address: ::c_ulong,
+    }
+}
+
+s_no_extra_traits! {
+    pub struct ucontext_t {
+        pub uc_flags: ::c_ulong,
+        pub uc_link: *mut ucontext_t,
+        pub uc_stack: ::stack_t,
+        // NOTE This field's type is actually mcontext_t,
+        //      which is a synonym for sigcontext.
+        //      But to avoid the error reported by style checker,
+        //      it must be sigcontext.
+        pub uc_mcontext: sigcontext,
+        // FIXME this and __padding_rt_sigset are actually a union
+        pub uc_sigmask: ::sigset_t,
```

Comment:
Then use an `union` type.

Reply:
How can I write `union`s like this in Rust?

```c
typedef struct ucontext {
  ...
  union {
    sigset_t uc_sigmask;
    sigset64_t uc_sigmask64;
  };
  ...
} ucontext_t;
```

(from https://android.googlesource.com/platform/bionic/+/refs/tags/ndk-r20/libc/include/sys/ucontext.h#381)

At first I made a separate types to represent the anonymous `union`s ([here](https://github.com/rust-lang/libc/pull/1622/commits/546a690047f01e85dd50f5b0967fa03be5294a3b#diff-8c71e51ad9afb09c731c4ec6ac99ed44R55-R58)):

```rust
            pub struct ucontext_t {
...
                pub uc_sigmask: uc_sigmask,
...
            }

            pub struct uc_sigmask_with_padding {
                pub uc_sigmask: ::sigset_t,
                /* Android has a wrong (smaller) sigset_t on x86. */
                __padding_rt_sigset: u32,
            }

            pub union uc_sigmask {
                uc_sigmask: uc_sigmask_with_padding,
                uc_sigmask64: ::sigset64_t,
            }
        }
```

But this was rejected by `libc-test` (Sorry I can't retrieve the log from CI).

As far as I know this isn't possible in current Rust.
I guessed that is why there are several comments such as `// FIXME this is actually a union` in this repository.

Reply:
> At first I made a separate types to represent the anonymous unions

That's how you do it.

> But this was rejected by libc-test (Sorry I can't retrieve the log from CI).

You probably just need to skip the union in the `libc-test/build.rs`, since it does not exist as a named union in C.